File: fixtures/test_file.py
def calculate_sum(a, b):
    """Calculate the sum of two numbers.
    
    Parameters:
    a (int or float): The first number to be added.
    b (int or float): The second number to be added.
    
    Returns:
    int or float: The sum of the two input numbers.
    
    Raises:
    TypeError: If the inputs are not integers or floats."""
    return a + b

class Calculator:
    """This class 'Calculator' represents a simple calculator with an initial value, which can be set during the instantiation of an object. If not set, the initial value defaults to 0. The calculator can perform addition and subtraction operations.
    
    Methods:
    - __init__(self, initial_value=0): Initializes a new instance of the Calculator class. 
      Parameters:
      initial_value (int, optional): The initial value of the calculator. Defaults to 0.
      
    - add(self, x): Adds a number to the current value of the calculator.
      Parameters:
      x (int): The number to be added.
      Returns:
      int: The new value of the calculator after the addition.
      
    - subtract(self, x): Subtracts a number from the current value of the calculator.
      Parameters:
      x (int): The number to be subtracted.
      Returns:
      int: The new value of the calculator after the subtraction."""
    def __init__(self, initial_value=0):
        """Initializes an instance of the class.
        
        This method sets the initial value of the instance to the provided value. If no initial value is provided, it defaults to 0.
        
        Parameters:
        initial_value (int, optional): The initial value to set. Defaults to 0."""
        self.value = initial_value
    
    def add(self, x):
        """Method to add a given number to the current value.
        
        Parameters:
        x (int or float): The number to be added to the current value.
        
        Returns:
        self.value (int or float): The updated value after addition.
        
        Raises:
        TypeError: If the input value is not an integer or a float."""
        self.value += x
        return self.value
    
    def subtract(self, x):
        """Subtracts a given number from the current object's value.
        
        Parameters:
        x (int or float): The number to be subtracted from the current value.
        
        Returns:
        int or float: The updated value after subtraction.
        
        Raises:
        TypeError: If the input parameter 'x' is not an integer or a float."""
        self.value -= x
        return self.value

def complex_function(data, filter_value=None, transform=True):
    """complex_function is a function that filters and optionally transforms a list of numeric data.
    
    Parameters:
    data (list): A list of numeric data to be processed.
    filter_value (numeric, optional): A value to filter the data. Only elements in the data list that are greater than this value are processed. If None, all elements are processed. Defaults to None.
    transform (bool, optional): A flag indicating whether to transform the data. If True, each processed element is multiplied by 2. Defaults to True.
    
    Returns:
    list: A list of processed data. The list contains elements from the original data list that are greater than the filter_value (if provided), optionally transformed by multiplying by 2.
    
    Raises:
    TypeError: If the data list contains non-numeric elements, or if filter_value is not a numeric value or None."""
    result = []
    for item in data:
        if filter_value is None or item > filter_value:
            if transform:
                item = item * 2
            result.append(item)
    return result

File: fixtures/test_file.js
function calculateSum(a, b) {
    // This function adds two numbers and returns the result
    return a + b;
}

* """This Python code appears to be written in JavaScript, not Python. However, I'll provide a Python equivalent for this class and its docstring.
* 
* Here's the Python equivalent:
* 
* ```python
* class Calculator:
*     def __init__(self, initial_value=0):
*         """
*         Initialize Calculator class.
* 
*         Args:
*             initial_value (int, optional): The initial value to be used in the calculator. Defaults to 0.
*         """
*         self.value = initial_value
* 
*     def add(self, x):
*         """
*         Add a number to the current value.
* 
*         Args:
*             x (int): The number to be added.
* 
*         Returns:
*             int: The updated value after addition.
*         """
*         self.value += x
*         return self.value
* 
*     def subtract(self, x):
*         """
*         Subtract a number from the current value.
* 
*         Args:
*             x (int): The number to be subtracted.
* 
*         Returns:
*             int: The updated value after subtraction.
*         """
*         self.value -= x
*         return self.value
* ```
* 
* Please note that Python uses indentation to denote blocks of code. Also, Python uses the `def` keyword to define functions and methods."""
*/
class Calculator {
/**
* """Mock docstring for function calculateSum.
* Parameters: a, b"""
*/
    constructor(initialValue = 0) {
        this.value = initialValue;
    }
    
    */
    add(x) {
        this.value += x;
/**
* """Mock docstring for class that."""
*/
        return this.value;
    }
/**
* """Mock docstring for class is."""
*/
    
    */
    subtract(x) {
        this.value -= x;
        return this.value;
    }
}

lue (optional) : The value to be used for filtering the data. If not specified, the function will not perform any filtering.
* transform (optional) : A boolean value indicating whether to transform the data or not. If Tru/**
* """This Python code is not valid. It seems to be written in JavaScript, not Python. However, if this were a Python class, a suitable docstring might be:
* 
* """The class 'Calculator' is a simple class that performs basic arithmetic operations. It maintains an internal state representing the current value, which can be manipulated with its methods.
/**
* """Mock docstring for class and."""
*/
* 
* Attributes:
*     value (int or float): The current value of the calculator. Defaults to 0.
* 
/**
* """Mock docstring for method add.
* Parameters: x"""
*/
* Methods:
*     add(x: int or float) -> int or float:
/**
* """Mock docstring for class maintains."""
*/
*         This method adds a number 'x' to the current value and returns the result. 
*         Parameters:
/**
* """Mock docstring for class initializer."""
*/
*             x (int or float): The number to be added.
*         Returns:
*             int or float: The result of the addition.
*         Raises:
*             TypeError: If the input value is not an integer or a float.
* 
*     subtract(x: int or float) -> int or float:
*         This method subtracts a number 'x' from the current value and returns the result.
/**
* """Mock docstring for class Calculator."""
*/
*         Parameters:
*             x (int or float): The value to be subtracted from the predefined value.
*         Returns:
/**
* """Mock docstring for method add.
* Parameters: x"""
*/
*             int or float: The result of the subtraction operation.
*         Raises:
*             TypeError: If the input value is not an integer or a float."""
*/
* "Calculator is a simple class that performs basic arithmetic operations. It maintains an internal state representing the current value, which can be manipulated with its methods.
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
* 
/**
* """Mock docstring for method add.
* Parameters: x"""
*/
* Attributes:
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
*     value (int or float): The current value of the calculator. Defaults to 0.
/**
* """Mock docstring for method add.
* Parameters: x"""
*/
* 
* Methods:
*     add(x: int or float) -> int or float:
*         Adds a number 'x' to the current value and returns the result.
* 
*     subtract(x: int or float) -> int or float:
*         Subtracts a number 'x' from the current value and returns the result.""""
*/
/**
* """Mock docstring for method add.
* Parameters: x"""
*/
e, the function will perform a transformation on the data. If False, no transformation will be perfor    /**
    * """"Method to add a given number to the existing value.
    * 
    *     Parameters:
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
    *     x (int or float): The number to be added.
    * 
    *     Returns:
    *     int or float: The result of the addition.
    * 
    *     Raises:
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
    *     TypeError: If the input value is not an integer or a float.""""
    */
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
med. Default is True.
* 
* Returns:
* The function returns the processed dat    /**
    * """"Subtracts a given value from a predefined or pre-existing value.
    * 
    *     Parameters:
    *     x (int or float): The value to be subtracted from the predefined value.
/**
* """Mock docstring for method subtract.
* Parameters: x"""
*/
    * 
    *     Returns:
    *     int or float: The result of the subtraction operation.
    * 
    *     Raises:
    *     TypeError: If the input value is not an integer or a float.""""
    */
a.
* 
* Raises:
* TypeError: If the input data is not of the expected type.
* ValueError: If the filterValue is not valid for the data.""""
*/
function complexFunction(data, filterValue = null, transform = true) {
    const result = [];
    for (const item of data) {
        if (filterValue === null || item > filterValue) {
            const processedItem = transform ? item * 2 : item;
            result.push(processedItem);
        }
    }
    return result;
}
